### ISSUE TITLE
Missing roles/appengine.deployer role but service account actually has it

### ISSUE THREAD

**Issue by Nallo** [google-github-actions/auth#256]
### TL;DR

Trying to Authenticate via Workload Identity Federation, created a Google Cloud service account and grant IAM permissions, created and configured a Workload Identity Provider for GitHub but `gcloud app deploy` is failing with

> Error: google-github-actions/deploy-appengine failed with: failed to execute gcloud command `gcloud app deploy --quiet --format json app.yaml --promote`: ERROR: (gcloud.app.deploy) Permissions error fetching application [apps/PROJED_ID]. Please make sure that you have permission to view applications on the project and that SERVICE_ACCOUT has the App Engine Deployer (roles/appengine.deployer) role.

How can I set the permissions below?

<img width="860" alt="Screenshot 2023-01-15 at 17 53 00" src="https://user-images.githubusercontent.com/3810408/212554564-5d10b370-efc1-49c5-9c60-568fe7efb5d2.png">

### Expected behavior

Deploy to App Engine

### Observed behavior

Command

> gcloud app deploy --quiet --format json app.yaml --promote

failing with

> Error: google-github-actions/deploy-appengine failed with: failed to execute gcloud command `gcloud app deploy --quiet --format json app.yaml --promote`: ERROR: (gcloud.app.deploy) Permissions error fetching application [apps/PROJED_ID]. Please make sure that you have permission to view applications on the project and that SERVICE_ACCOUT has the App Engine Deployer (roles/appengine.deployer) role.

### Action YAML

```yaml
# This is a basic workflow to help you get started with Actions

name: Deploy to GAE

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the main branch
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  deploy:
    name: Deploying to Google Cloud
    runs-on: ubuntu-latest

    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
    - name: Checkout
      uses: actions/checkout@v3

    - name: Authenticate
      id: auth
      uses: google-github-actions/auth@v1
      with:
        create_credentials_file: true
        workload_identity_provider: 'projects/XXXXXX/locations/global/workloadIdentityPools/github-pool/providers/YYYYYYY'
        service_account: 'YYYYYY@ZZZZZZ.iam.gserviceaccount.com'

    - name: Deploy to App Engine
      id: deploy
      uses: google-github-actions/deploy-appengine@v1

    - name: Test
      run: curl "${{ steps.deploy.outputs.url }}"
```

### Log output

```text
Run google-github-actions/deploy-appengine@v1
  with:
    promote: true
  env:
    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/gae-expenses-report/gae-expenses-report/gha-creds-8e7d37d3432eaea1.json
    GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/gae-expenses-report/gae-expenses-report/gha-creds-8e7d37d3432eaea1.json
    GOOGLE_GHA_CREDS_PATH: /home/runner/work/gae-expenses-report/gae-expenses-report/gha-creds-8e7d37d3432eaea1.json
    CLOUDSDK_CORE_PROJECT: YYYYYY
    CLOUDSDK_PROJECT: YYYYYY
    GCLOUD_PROJECT: YYYYYY
    GCP_PROJECT: YYYYYY
    GOOGLE_CLOUD_PROJECT: YYYYYY
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/016a0ad7-6fd1-4a1e-bdb5-fb2198257f3e -f /home/runner/work/_temp/0d900d25-2837-48d4-bd95-879be82daa8d
Successfully authenticated
Running: gcloud app deploy --quiet --format json app.yaml --promote
Error: google-github-actions/deploy-appengine failed with: failed to execute gcloud command `gcloud app deploy --quiet --format json app.yaml --promote`: ERROR: (gcloud.app.deploy) Permissions error fetching application [apps/YYYYYY]. Please make sure that you have permission to view applications on the project and that YYYYYY has the App Engine Deployer (roles/appengine.deployer) role.
```

### Additional information

From App Engine console:

Service account principals

<img width="1412" alt="Screenshot 2023-01-15 at 17 46 41" src="https://user-images.githubusercontent.com/3810408/212554366-a2fcd297-7270-465d-a4b1-44c2a8db60aa.png">

Principal set has following roles

<img width="732" alt="Screenshot 2023-01-15 at 17 49 25" src="https://user-images.githubusercontent.com/3810408/212554444-9e2d13d1-11e6-4291-92e0-ab13b5e3c432.png">

Custom Role has following App Engine permissions

<img width="269" alt="Screenshot 2023-01-15 at 17 51 42" src="https://user-images.githubusercontent.com/3810408/212554516-c8da986d-743c-4618-a47f-4c56c4a843b1.png">
